src/utils/logging.rs

```diff
@@ -1,30 +1,66 @@
-use std::io;
-use std::io::Write;
-use std::mem;
-use std::sync::atomic::{AtomicUsize, Ordering};
-use std::sync::{Arc, Weak};
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc, Weak,
+};
 
 use chrono::Local;
 use console::{style, Color};
 use indicatif::ProgressBar;
 use lazy_static::lazy_static;
+use log::max_level;
 use parking_lot::RwLock;
 
+// Quiet mode decides whether to print something based on `--quiet` flag.
+lazy_static! {
+    static ref QUIET_MODE: AtomicBool = AtomicBool::new(false);
+}
+
+pub fn quiet_mode() -> bool {
+    QUIET_MODE.load(Ordering::Relaxed)
+}
+
+pub fn set_quiet_mode(yes: bool) {
+    QUIET_MODE.store(yes, Ordering::Relaxed);
```

Comment:
nit:

```suggestion
pub fn set_quiet_mode(is_quiet: bool) {
    QUIET_MODE.store(is_quiet, Ordering::Relaxed);
```

Reply:
I took this convenvtion from clap, but I agree it's better with `is_quiet`